src/stack.jl

```diff
@@ -17,13 +17,12 @@ function push!(s::Stack, x)
     s
 end
 
-#returns a collection that can be used in a for loop
-function iter{T}(s::Stack{T})
-    a = T[]
-    for i in s.store
-        unshift!(a,i)
-    end
-    return a
-end
-
 pop!(s::Stack) = pop!(s.store)
+
+start(st::Stack) = start_reverse(st.store)
+next(st::Stack, s) = next_reverse(st.store, s)
+done(st::Stack, s) = done_reverse(st.store, s)
```

Comment:
I think I would just do

```julia
start(st::Stack) = start(reverse_iter(st.store))
next(st::Stack, s) = next(reverse_iter(st.store), s)
done(st::Stack, s) = done(reverse_iter(st.store), s)
```

and then remove `start_reverse`, `next_reverse`, and `done_reverse`.

The compiler should inline these, so they should produce the same code.